### ISSUE TITLE
Document canonical way to set priority for static pods

### ISSUE THREAD

**Issue by stackbaek** [kubernetes/website#46859]
Per [doc](https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#pod-selection-for-kubelet-eviction:~:text=Static%20pods%20do%20not%20support%20the%20priorityClassName%20field.), static pods does not support priorityClassName field. However, when static pods are created with only `priority` values set to `2000001000` (that matches with  `system-node-critical`) kubelet prints the following error:

```
forbidden: the integer value of priority (2000001000) must not be provided in pod spec; priority admission controller computed 0 from the given PriorityClass name" pod="kube-system/...
```

**Comment by thisisharrsh** [comment]
Page related to the issue: https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#pod-selection-for-kubelet-eviction

/sig node

**Comment by sftim** [comment]
What's the defect @stackbaek? Are you saying that the docs are incorrect, that there is a bug in the kubelet, or that there is a problem but you are not sure where?

**Comment by stackbaek** [comment]
The instruction to add priority to static pods are unclear from the doc. From the doc, it implies that I should just add `spece.priority` to the manifest (since `priorityClassName` was documented to be [not supported](https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#pod-selection-for-kubelet-eviction:~:text=Static%20pods%20do%20not%20support%20the%20priorityClassName%20field.)). In reality, I had to add both `priority` and `priorityClassName`. 

At the end of the day, I just want to confirm the canonical ways to set priority to the static pods to ensure node critical pods are running under node pressure.

---

NOTE: I have not tried to set just `priorityClassName` on the static pods.

**Comment by sftim** [comment]
/retitle Document canonical way to set priority for static pods

**Comment by thisisharrsh** [comment]
The necessary information needs to be added, and the documentation should be corrected. I'd want to hear from more people too.

**Comment by ktvargo-ms** [comment]
@SergeyKanzhelev I was able to create a static pod with priorityClassName and was planning on adding the example to the document.  Let me know if anything else is needed or if that isn't expected behavior.  I see you have this added for triage with Node bugs.  

k get priorityclass
NAME                      VALUE        GLOBAL-DEFAULT   AGE
system-cluster-critical   2000000000   false            5d2h
system-node-critical      2000001000   false            5d2h


# cat static-web.yaml 
apiVersion: v1
kind: Pod
metadata:
  name: static-web
  labels:
    role: myrole
spec:
  containers:
    - name: web
      image: nginx
      ports:
        - name: web
          containerPort: 80
          protocol: TCP
  priorityClassName: system-cluster-critical


k describe pod static-web-minikube	

Name:                 static-web-minikube
Namespace:            default
Priority:             2000000000
Priority Class Name:  system-cluster-critical
Node:                 minikube/192.168.49.2
Start Time:           Tue, 25 Jun 2024 15:18:25 -0400
Labels:               role=myrole
Annotations:          kubernetes.io/config.hash: d13d717ea915d4c4049f7783bb26358a
                      kubernetes.io/config.mirror: d13d717ea915d4c4049f7783bb26358a
                      kubernetes.io/config.seen: 2024-06-25T19:18:25.377885764Z
                      kubernetes.io/config.source: file
Status:               Running

**Comment by ktvargo-ms** [comment]
@stackbaek and @sftim, I was able to confirm that using just priorityClassName on the static pods works (see example above).  Since this is a concept page, we do want to add an example or not?  Otherwise, it is just a quick update on the docs to use  priorityClassName on the static pods as well.

**Comment by stackbaek** [comment]
@ktvargo-ms great! examples always boosts clarity and readability (at least for me). 

So, I guess we don't even need the priority number, correct? Otherwise all SGTM!

**Comment by sftim** [comment]
> @stackbaek and @sftim, I was able to confirm that using just priorityClassName on the static pods works (see example above). Since this is a concept page, we do want to add an example or not? Otherwise, it is just a quick update on the docs to use priorityClassName on the static pods as well.

Interesting (and to me surprising); however, let's show that this works for an arbitrary Pod priority and not just a built-in one. We should also show that it'll work OK for a static Pod that starts before the control plane is healthy.

**Comment by ktvargo-ms** [comment]
The original issue (improper instruction) has been resolved.  Would these items be better addressed in the static pod page?

**Comment by ktvargo-ms** [comment]
> @ktvargo-ms great! examples always boosts clarity and readability (at least for me).
> 
> So, I guess we don't even need the priority number, correct? Otherwise all SGTM!

Correct. Only the priorityClassName is required.   The priority number is displayed when describing the node.

**Comment by sftim** [comment]
> Only the priorityClassName is required. The priority number is displayed when describing the node.

I don't get how this works if the control plane isn't up yet. That's an important consideration.

**Comment by ktvargo-ms** [comment]
Created 'xyz-priority' and will use that an example

k get priorityclass
NAME                      VALUE        GLOBAL-DEFAULT   AGE
system-cluster-critical   2000000000   false            2m2s
system-node-critical      2000001000   false            2m2s
xyz-priority              1000000      false            47s

apiVersion: v1
kind: Pod
metadata:
  name: static-web
  labels:
    role: myrole
spec:
  containers:
    - name: web
      image: nginx
      ports:
        - name: web
          containerPort: 80
          protocol: TCP
  priorityClassName: xyz-priority

k describe pod static-web-minikube
Name:                 static-web-minikube
Namespace:            default
Priority:             1000000
Priority Class Name:  xyz-priority
Node:                 minikube/192.168.49.2
Start Time:           Tue, 02 Jul 2024 15:56:35 -0400
Labels:               role=myrole
Annotations:          kubernetes.io/config.hash: a8236a64037ee18291f146e8a8013147
                      kubernetes.io/config.mirror: a8236a64037ee18291f146e8a8013147
                      kubernetes.io/config.seen: 2024-07-02T19:56:35.763567682Z
                      kubernetes.io/config.source: file
Status:               Running

**Comment by AnishShah** [comment]
sig-node meeting notes:

/triage accepted
